### PR TITLE
PYTHON-2914 MongoClient should raise an error when given multiple URIs

### PR DESCRIPTION
**Pull request by juliusgeo**

### CODE REVIEW

File: pymongo/mongo_client.py [hunk 1]
```diff
@@ -645,6 +645,10 @@ def __init__(
         dbase = None
         opts = common._CaseInsensitiveDictionary()
         fqdn = None
+        # Raise an error when we have more than one URI
+        if len([i for i in host if "/" in i]) > 1:
+            raise ConfigurationError("MongoClient() only accepts 1 URI at a "
+                                     "time")
```
Comment (ShaneHarvey) [inline pymongo/mongo_client.py:651]: I'd say: "host must not contain multiple MongoDB URIs"

File: pymongo/mongo_client.py [hunk 2]
```diff
@@ -645,6 +645,10 @@ def __init__(
         dbase = None
         opts = common._CaseInsensitiveDictionary()
         fqdn = None
+        # Raise an error when we have more than one URI
```
Comment (ShaneHarvey) [inline pymongo/mongo_client.py:648]: This comment is redundant because the error message says the same thing.

File: pymongo/mongo_client.py [hunk 3]
```diff
@@ -645,6 +645,10 @@ def __init__(
...
+        if len([i for i in host if "/" in i]) > 1:
```
Comment (ShaneHarvey) [inline pymongo/mongo_client.py:649]: i->h

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
Please also update the MongoClient docs to remove the parts about accepting multiple URIs.

File: doc/migrate-to-pymongo4.rst [hunk 4]
```diff
@@ -187,6 +187,12 @@ MongoClient cannot execute operations after ``close()``
 after being closed. The previous behavior would simply reconnect. However,
 now you must create a new instance.
 
+MongoClient raises exception when given more than 1 URI
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply doc/migrate-to-pymongo4.rst]: Done.

File: pymongo/mongo_client.py (hunk 2 above)
Comment (juliusgeo) [reply pymongo/mongo_client.py]: Done.

File: pymongo/mongo_client.py (hunk 3 above)
Comment (juliusgeo) [reply pymongo/mongo_client.py]: Done.

File: pymongo/mongo_client.py (hunk 1 above)
Comment (juliusgeo) [reply pymongo/mongo_client.py]: Done.

File: pymongo/mongo_client.py [hunk 5]
```diff
@@ -109,11 +109,9 @@ def __init__(
...
         The `host` parameter can be a full `mongodb URI
         <http://dochub.mongodb.org/core/connections>`_, in addition to
-        a simple hostname. It can also be a list of hostnames or
-        URIs. Any port specified in the host string(s) will override
-        the `port` parameter. If multiple mongodb URIs containing
-        database or auth information are passed, the last database,
-        username, and password present will be used.  For username and
+        a simple hostname. It can also be a list of hostnames but no more
+        than one URI. Any port specified in the host string(s) will override
+        the `port` parameter. For username and
```
Comment (ShaneHarvey) [inline pymongo/mongo_client.py:114]: We need to make a similar change to the description in the `:Parameters:` list below.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: pymongo/mongo_client.py (hunk 5 above)
Comment (juliusgeo) [reply pymongo/mongo_client.py:114]: Done.

**Review by ShaneHarvey — APPROVED** [review]
LGTM

**Pull request merged**